builder/proxmox/common/step_stop_vm.go

```diff
@@ -0,0 +1,39 @@
+package proxmox
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Telmate/proxmox-api-go/proxmox"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// stepStopVM takes the running VM configured in earlier steps and stops it.
+
+type stepStopVM struct{}
+
+type vmTerminator interface {
+	ShutdownVm(*proxmox.VmRef) (string, error)
+}
+
+var _ templateConverter = &proxmox.Client{}
+
+func (s *stepStopVM) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
```

Comment:
I wonder why this becomes a step on its own here, it seems that the `step_convert_to_template` does this already, why not move the check after the VM is shutdown so we don't have to introduce another step?
For the record, I don't hate this change by itself, it's just that this brings more changes to this PR which makes the diff larger and harder to understand.